app/components/SignatureCard/index.tsx

```diff
@@ -36,23 +39,41 @@ const SignatureCard = ({
   type,
   isPrint = false,
 }: SignatureCardProps): JSX.Element => {
-  const { canEdit } = useReport();
+  const { reportAssignmentAccess: canAddSignatures } = useResource();
+  const { canEdit, report, setReport } = useReport();
+  const { userDetails } = useSecurity();
+
   const [userSignature, setUserSignature] = useState<UserType>();
+  const [role, setRole] = useState('');
 
   useEffect(() => {
     if (signatures && type) {
       if (type === 'author') {
         setUserSignature(signatures.authorSignature);
+        setRole('analyst');
       } else if (type === 'reviewer') {
         setUserSignature(signatures.reviewerSignature);
+        setRole('reviewer');
       } else if (type === 'creator') {
         setUserSignature(signatures.creatorSignature);
+        setRole('bioinformatician');
       }
     }
-  }, [signatures, type]);
+  }, [signatures, type, setRole]);
 
-  const handleSign = () => {
-    onClick(true, type);
+  const handleSign = async () => {
+    try {
+      const newReport = await api.post(
```

Comment:
separating the API calls on PUT requests to `report` and `signatures` on different components will cause discrepencies on the UI whenever a user interacts with "sign".

This is actually a good idea to put the API calls on the card itself, let's update the signatures here instead and send the callback for `onClick(true, type)` whenever both calls succeed, and we can also update the report here as well, instead of 
`const handleSign = useCallback(async (signed: boolean, role: SignatureUserType) ` we can have
`const handleSign = useCallback(async (signed: boolean, updatedSignatures)` instead